Source/WebCore/dom/Document.cpp

```diff
@@ -2090,8 +2090,6 @@ void Document::updateRenderTree(std::unique_ptr<const Style::Update> styleUpdate
 
 void Document::resolveStyle(ResolveStyleType type)
 {
-    ScriptDisallowedScope::InMainThreadOfWebProcess scriptDisallowedScope;
```

Comment:
The top level one goes away because all callers have the scope already? Maybe we should assert that?

Reply:
Yeah, `Document::updateStyle` already has `ScriptDisallowedScope::InMainThread`.
Sure, we can assert `ASSERT(ScriptDisallowedScope::InMainThread::hasDisallowedScope());`

Reply:
Turns out there are some callers without this. Adding in-main-thread variant back.